### PR TITLE
Move metadata to pyproject.toml

### PR DESCRIPTION
**Pull request by cdce8p**
## What do these changes do?

The metadata changes:

```diff
 ...
-Maintainer: aiohttp team <team@aiohttp.org>
-Maintainer-email: team@aiohttp.org
+Maintainer-email: aiohttp team <team@aiohttp.org>
 License: Apache-2.0 AND MIT
-Home-page: https://github.com/aio-libs/aiohttp
+Project-URL: Homepage, https://github.com/aio-libs/aiohttp
 ...
```

**Open questions**
- Should this PR be backported? It's possible, might just require some manual work due to merge conflicts.

### CODE REVIEW

**Comment by webknjaz** [comment]
Let's not rush this. Keeping a dedicated config is a conscious choice. Also, PRs must not reformat half a project along with changing the packaging configs.

FWIW I'd rather keep it as is for now. Especially, since I was going to sync the configs across the projects and this is going to cause a lot of headache since it conflicts with how things are now.

I'm 👎 on this migration right now. It'd be interesting to gather the current dependency limitations in the downstreams to make an informed choice. Additionally, the move would have to happen in the dependencies first.

**Review by asvetlov — COMMENTED** [review]
I personally don't care where the project metadata sits until it works and I personally shouldn't spend additional time on the infrastructure maintenance.

**Comment by asvetlov** [comment]
> That's actually a bug with the current black config.

@cdce8p could you please extract a fix for this particular problem into a separate PR? A small patch has a great chance to  be reviewed and approved quickly.

**Comment by asvetlov** [comment]
Ooh, sorry. I've pushed the wrong button and closed the PR. Please accept my apologizes.

**Comment by cdce8p** [comment]
> > That's actually a bug with the current black config.
> 
> @cdce8p could you please extract a fix for this particular problem into a separate PR? A small patch has a great chance to be reviewed and approved quickly.

See #9962. It looks bigger than it actually is.

**Comment by asvetlov** [comment]
Thanks, @cdce8p 
The PR is merged and back ported

**Comment by cdce8p** [comment]
Rebased the PR. Also moved the `dependencies` and `optional-dependencies` back to `setup.cfg` to purely focus on the static metadata. Please let me know if you're interested in moving this forward. If not or not at this time, feel free to close it.

File: setup.cfg [hunk 1]
```diff
@@ -1,55 +1,4 @@
...
 [options]
-python_requires = >=3.9
-packages =
-  aiohttp
-  aiohttp._websocket
-# https://setuptools.readthedocs.io/en/latest/setuptools.html#setting-the-zip-safe-flag
-zip_safe = False
-include_package_data = True
-
 install_requires =
```
Comment (Dreamsorcerer) [inline setup.cfg:2]: I would have thought the dependencies would go in the same file as the rest of the metadata..?

Comment (cdce8p) [reply setup.cfg:2]: Yes they can be moved as well. Just the current tooling expects them to be in `setup.cfg`. Since this PR stalled for so long I figured I'd revert that part for now. See my last commit. https://github.com/aio-libs/aiohttp/pull/9951/commits/ec9022547a07f710a83a5c3e172884ccc82632bb

_I'd be happy to re-include it here or do it in a followup if there is interest._

Comment (Dreamsorcerer) [reply setup.cfg:2]: Right. I'll defer to @webknjaz here to decide how it should go.

File: pyproject.toml [hunk 2]
```diff
@@ -1,9 +1,57 @@
 [build-system]
-requires = [
-    "setuptools >= 46.4.0",
-]
+requires = ["setuptools >= 70.0.0"]
 build-backend = "setuptools.build_meta"
 
+[project]
```
Comment (asvetlov) [inline pyproject.toml:12]: Does dependabot understand this format?
I know that dependabot for poetry still requires the old `[tool.poetry]` section instead of the recommended `[project]`, I wonder if the setuptools-based check is different.

Comment (cdce8p) [reply pyproject.toml:12]: > Does dependabot understand this format?

Yes, that's a poetry only bug in dependabot.

Comment (webknjaz) [reply pyproject.toml:12]: I don't think dependabot updates abstract deps anyway. Renovate probably would. Though, in general, this shouldn't be updated automatically. It's not a lock file and mustn't pin things or contain top boundaries in general. This is something that pip-tools constraints would solve, as they'd lock envs w/o the dangers of causing problems in dependency resolution for the end users.

Comment (cdce8p) [reply pyproject.toml:12]: As mentioned earlier, I reverted the part that moved the dependencies to only focus on the static metadata here.

**Comment by webknjaz** [comment]
> Rebased the PR. Also moved the `dependencies` and `optional-dependencies` back to `setup.cfg` to purely focus on the static metadata. Please let me know if you're interested in moving this forward. If not or not at this time, feel free to close it.

Not sure if this would work well with `setuptools`. I think when it detects PEP 621 mode, it would only source the spec-compliant bits from `pyproject.toml`. There might be some backend configs that would be sourced from other places, but anything that's not listed as dynamic explicitly is likely to be disregarded or otherwise cause breakage. So with this move we'll need to migrate things atomically.

File: pyproject.toml (hunk 2 above)
Comment (webknjaz) [reply pyproject.toml:12]: Ah, I didn't notice that you marked those as dynamic.

File: pyproject.toml [hunk 3]
```diff
@@ -1,9 +1,57 @@
...
+"Docs: Changelog"    = "https://docs.aiohttp.org/en/stable/changes.html"
+"Docs: RTD"          = "https://docs.aiohttp.org"
+"GitHub: issues"     = "https://github.com/aio-libs/aiohttp/issues"
+"GitHub: repo"       = "https://github.com/aio-libs/aiohttp"
+
+[tool.setuptools.dynamic]
+version = {attr = "aiohttp.__version__"}
+
+[tool.setuptools.packages.find]
+include = ["aiohttp*"]
```
Comment (webknjaz) [inline pyproject.toml]: This will probably happen automatically.

File: pyproject.toml (hunk 2 above)
Comment (webknjaz) [reply pyproject.toml:12]: But still, there's no need to separate those anyway. So let's get them back into this config.

File: setup.cfg [hunk 4]
```diff
@@ -1,55 +1,4 @@
...
-
-  Topic :: Internet :: WWW/HTTP
-
 [options]
-python_requires = >=3.9
-packages =
-  aiohttp
-  aiohttp._websocket
-# https://setuptools.readthedocs.io/en/latest/setuptools.html#setting-the-zip-safe-flag
-zip_safe = False
```
Comment (webknjaz) [inline setup.cfg:50]: This is no longer necessary?

File: pyproject.toml [hunk 5]
```diff
@@ -1,9 +1,57 @@
...
+exclude = ["examples"]
+
+[tool.setuptools.package-data]
+"*" = ["*.so"]
+
+[tool.setuptools.exclude-package-data]
+"*" = ["*.c", "*.h"]
```
Comment (webknjaz) [inline pyproject.toml]: I'm not entirely sure if these are still needed. Might need some manual checks.

File: setup.cfg (hunk 1 above)
Comment (webknjaz) [reply setup.cfg:2]: It should move to `pyproject.toml`. The tooling should be updated.

File: CHANGES/9951.packaging.rst [hunk 6]
```diff
@@ -0,0 +1 @@
+Moved static metadata from ``setup.cfg`` to ``pyproject.toml`` -- by :user:`cdce8p`.
```
Comment (webknjaz) [inline CHANGES/9951.packaging.rst]:
```suggestion
Moved core packaging metadata from :file:`setup.cfg` to :file:`pyproject.toml` per :pep:`621`
-- by :user:`cdce8p`.
```
(it's not really just static)

File: pyproject.toml [hunk 7]
```diff
@@ -1,9 +1,57 @@
 [build-system]
-requires = [
-    "setuptools >= 46.4.0",
-]
+requires = ["setuptools >= 70.0.0"]
```
Comment (webknjaz) [inline pyproject.toml]: What's the specific feature that requires at least v70?

**Comment by webknjaz** [comment]
> > I'm 👎 on this migration right now. It'd be interesting to gather the current dependency limitations in the downstreams to make an informed choice. Additionally, the move would have to happen in the dependencies first.
> 
> I'm probably missing something here. This PR doesn't affect any downstream projects. The project metadata change is minimal and spec compliant.
> 
> Tbh I just thought it might be a good time to move the static metadata after I learned that it can work fine with `setup.py`. It will probably still take a while (couple of years) but I expect setuptools to eventually deprecate the `setup.cfg` config.

What I meant back then was that I'd like the move to first happen in projects that are direct deps of aiohttp (like yarl). This is because I've implemented some advanced packaging techniques there (like an in-tree build backend). I'm going to eventually migrate aiohttp to use the same model. So packaging updates would flow from those places where improvements were first implemented.

Upon reflection, thought, I think that this PR doesn't block that so you're fine. Talking about the downstreams, they've been historically behind some upstream packaging standards, but AFAIK they've mostly implemented their own PEP 517 helpers by now. Still, there might be problems where the setuptools version they ship is older than our spec (it'd conflict, forcing them to patch it out on their side). Also, if they patch it out, there's a possibility that it's older than PEP 621 effectively not producing most of the metadata. So we should at least be careful about https://github.com/aio-libs/aiohttp/pull/9951/files#r2066217591.

**Comment by cdce8p** [comment]
That's for the review @webknjaz! I'd address all comments in one to keep it readable. Please let me know if you have any followup questions.

> Move dependencies to `pyproject.toml`

Sounds good. I'd just recommend to do that in a followup PR. Otherwise the backports PRs will likely not apply cleanly. In the end it's just a revert of https://github.com/aio-libs/aiohttp/pull/9951/commits/ec9022547a07f710a83a5c3e172884ccc82632bb

With `dependencies` and `optional-dependencies` marked as `dynamic`, they are still read from `setup.cfg` as well. Checked the generated `METADATA` file again just to be sure.

> ```toml
> [tool.setuptools.packages.find]
> include = ["aiohttp*"]
> ```
> This will probably happen automatically.

Unfortunately not. Setuptools will throw an error `Multiple top-level packages discovered in a flat-layout: ['vendor', 'aiohttp', 'CHANGES']`. We can get rid of `exclude = ["examples"]` though.

> ```ini
> zip_safe = False
> ```
> This is no longer necessary?

Yes. From the setuptools docs
> Obsolete - only relevant for pkg_resources, easy_install and setup.py install in the context of eggs (deprecated).

https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration

> > `package-data` and `exclude-package-data`
>
> I'm not entirely sure if these are still needed. Might need some manual checks.

`package-data` can indeed be removed. The default for `pyproject.toml` configs is `include-package-data = true`. We still need `exclude-package-data` though. Otherwise the wheels would include `*.c` and `*.h` files. I plan to do a followup to optimize the sdist / wheel sizes a bit by extending the list further. E.g. we still include the raw cython `*.pyx` files in the wheels which probably aren't necessary.

> What's the specific feature that requires at least v70?

Double checked that. It's actually `>= 66.1` that's required, so I'd suggest `>= 67.0`. Previous versions relied on `pkgutil.ImpImporter` which isn't available on Python 3.12+.

https://setuptools.pypa.io/en/latest/history.html#v66-1-0

> Still, there might be problems where the setuptools version they ship is older than our spec

Yes, I've seen a few reports in the light of the PEP 639 adoption. Support for it was only added in `v77`. I believe most (?) distributors still use an older version with no build isolation so that caused some confusion. That's one of the reasons I haven't updated the metadata here yet. `v67.0` should be fine.

**Comment by cdce8p** [comment]
Pushed a new commit to incorporate the license metadata changes from #11226.

@webknjaz Would you mind taking a look at this again? I believe I've addressed all your comments in https://github.com/aio-libs/aiohttp/pull/9951#issuecomment-2841603641. Please let me know if you have any more questions.

File: MANIFEST.in
Comment (webknjaz) [inline MANIFEST.in:1]: I'd rather not make unrelated changes. The change note talks about moving the metadata definitions, but the patch has some random changes without any justification. Let's not do this.

File: setup.cfg [hunk 8]
```diff
@@ -1,55 +1,4 @@
...
-  Docs: Changelog = https://docs.aiohttp.org/en/stable/changes.html
-  Docs: RTD = https://docs.aiohttp.org
-  GitHub: issues = https://github.com/aio-libs/aiohttp/issues
-  GitHub: repo = https://github.com/aio-libs/aiohttp
-description = Async http client/server framework (asyncio)
-long_description = file: README.rst
-long_description_content_type = text/x-rst
-maintainer = aiohttp team <team@aiohttp.org>
-maintainer_email = team@aiohttp.org
-license = Apache-2.0 AND MIT
```
Comment (webknjaz) [inline setup.cfg:19]: This SPDX identifier got lost somewhere.

File: pyproject.toml [hunk 9]
```diff
@@ -1,10 +1,62 @@
...
 requires = [
     "pkgconfig",
-    "setuptools >= 46.4.0",
+    "setuptools >= 67.0",
 ]
 build-backend = "setuptools.build_meta"
 
+[project]
+name        = "aiohttp"
+license     = {text = "Apache-2.0 AND MIT"}
```
Comment (webknjaz) [inline pyproject.toml:15]: This is incorrect. SPDX must not be added as license text: https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license

File: MANIFEST.in
Comment (cdce8p) [reply MANIFEST.in:1]: Could you specify which parts you're referring to? AFICT all changes are related / a result of the metadata move.
E.g. `tool.black.target-version` can be removed now that `project.requires-python` is specified.

**Comment by webknjaz** [comment]
> I believe I've addressed all your comments in [#9951 (comment)](https://github.com/aio-libs/aiohttp/pull/9951#issuecomment-2841603641). Please let me know if you have any more questions.

All other threads need to be addressed as well. There's a number of those with zero response.

File: setup.cfg (hunk 8 above)
Comment (cdce8p) [reply setup.cfg:19]: It's included as part of `project.license.text`.

File: pyproject.toml (hunk 9 above)
Comment (cdce8p) [reply pyproject.toml:15]: This is only supported in setuptools `v77` which most (?) distributors don't support just yet. That's why I've used `license.text` which is map to the same metadata as the old `license` field.

See also my longer comment about it: https://github.com/aio-libs/aiohttp/pull/9951#issuecomment-2841603641

**Comment by cdce8p** [comment]
> > I believe I've addressed all your comments in [#9951 (comment)](https://github.com/aio-libs/aiohttp/pull/9951#issuecomment-2841603641). Please let me know if you have any more questions.
> 
> All other threads need to be addressed as well. There's a number of those with zero response.

I've addressed them in https://github.com/aio-libs/aiohttp/pull/9951#issuecomment-2841603641 as I personally find the Github code comments hard to follow when it's a discussion about one or more related topics.

File: MANIFEST.in
Comment (webknjaz) [reply MANIFEST.in:1]: Every change in _this_ file.

File: pyproject.toml [hunk 10]
```diff
@@ -1,10 +1,62 @@
 [build-system]
 requires = [
     "pkgconfig",
-    "setuptools >= 46.4.0",
+    "setuptools >= 67.0",
```
Comment (webknjaz) [inline pyproject.toml:8]: Could you add a code comment explaining what feature needs this minimum version?

File: pyproject.toml [hunk 11]
```diff
@@ -1,10 +1,62 @@
...
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
+  "Topic :: Internet :: WWW/HTTP",
+]
+requires-python = ">=3.9"
```
Comment (webknjaz) [inline pyproject.toml]: Keep this sparse
```suggestion
requires-python = ">= 3.9"
```

**Comment by webknjaz** [comment]
> I believe I've addressed all your comments in [#9951 (comment)](https://github.com/aio-libs/aiohttp/pull/9951#issuecomment-2841603641).

That's why I missed it... Inline comments are attached to context they discuss. Dumping a bunch of text in an unrelated location make it more difficult to match with the actual diff. It's also something that makes it impossible to review on mobile. I'll have to find time to open on the computer and then I'll be able to actually read it in context with an extra window on screen.

File: MANIFEST.in
Comment (cdce8p) [reply MANIFEST.in:1]: setuptools includes `*.pyi` and `py.typed` files by default which made `global-include aiohttp *.pyi` redundant since `graft aiohttp` is also given. global-include also has the negative side effect of matching any aiohttp files inside venvs.

`recursive-include vendor *` and `graft vendor` are basically the same though I'd argue the graft command is easier to understand.

Will revert the change for `prune **/.hash`.

--
https://setuptools.pypa.io/en/latest/userguide/miscellaneous.html

File: pyproject.toml (hunk 2 above)
Comment (cdce8p) [reply pyproject.toml:12]: > But still, there's no need to separate those anyway. So let's get them back into this config.

Sounds good. I'd just recommend to do that in a followup PR. Otherwise the backports PRs will likely not apply cleanly. In the end it's just a revert of https://github.com/aio-libs/aiohttp/commit/ec9022547a07f710a83a5c3e172884ccc82632bb.

Happy to open a PR for it after this is merged.

File: pyproject.toml (hunk 3 above)
Comment (cdce8p) [reply pyproject.toml]: Unfortunately not. Setuptools will throw an error `Multiple top-level packages discovered in a flat-layout: ['vendor', 'aiohttp', 'CHANGES']`.

File: setup.cfg (hunk 4 above)
Comment (cdce8p) [reply setup.cfg:50]: `include_package_data = True` is the default for `pyproject.toml` based configs with setuptools.
`zip_safe = False` is considered obsolete. From the docs:
> Obsolete - only relevant for pkg_resources, easy_install and setup.py install in the context of eggs (deprecated).

https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration

File: setup.cfg (hunk 1 above)
Comment (cdce8p) [reply setup.cfg:2]: As mentioned earlier, I'd recommend to do that in a separate followup PR. Happy to do that after this one is merged.

**Review by cdce8p — COMMENTED** [review]
> > I believe I've addressed all your comments in [#9951 (comment)](https://github.com/aio-libs/aiohttp/pull/9951#issuecomment-2841603641).
> 
> That's why I missed it... Inline comments are attached to context they discuss. Dumping a bunch of text in an unrelated location make it more difficult to match with the actual diff. It's also something that makes it impossible to review on mobile. I'll have to find time to open on the computer and then I'll be able to actually read it in context with an extra window on screen.

Sorry about that. It seems our Github styles accidentally collided making it more difficult than it should have been. I've tried to respond to all open code comments now. Please feel free to resolve those which are done or highlight the ones I missed.

Will also push a minor commit shortly based on your suggestions.

File: setup.cfg (hunk 8 above)
Comment (webknjaz) [reply setup.cfg:19]: @cdce8p have you re-checked if this affects the metadata after all the updates applied recently?

File: pyproject.toml [hunk 12]
```diff
@@ -1,10 +1,66 @@
 [build-system]
 requires = [
     "pkgconfig",
-    "setuptools >= 46.4.0",
+    # setuptools >= 67.0 required for Python 3.12+ support
+    # Next step should be >= 77.0 for PEP 639 support
+    # Don't bump to early to give distributors time to update
```
Comment (webknjaz) [inline pyproject.toml]: typo?
```suggestion
    # Don't bump too early to give distributors time to update
```

File: pyproject.toml (hunk 10 above)
Comment (webknjaz) [reply pyproject.toml:8]: Marking this thread as resolved.

File: pyproject.toml (hunk 2 above)
Comment (webknjaz) [reply pyproject.toml:12]: Ack. Follow-up is good. Marking the thread as resolved.

File: pyproject.toml (hunk 9 above)
Comment (webknjaz) [reply pyproject.toml:15]: Could you add a code comment with the update guidelines, then?
And maybe multi-line this by separating it into `[project.license]`: I like it when it's sparse and everything is on dedicated lines as it helps with nicer diffs over time.

File: pyproject.toml [hunk 13]
```diff
@@ -1,10 +1,66 @@
...
+    "setuptools >= 67.0",
 ]
 build-backend = "setuptools.build_meta"
 
+[project]
+name        = "aiohttp"
+license     = {text = "Apache-2.0 AND MIT"}
+description = "Async http client/server framework (asyncio)"
+readme      = "README.rst"
+maintainers = [{name = "aiohttp team", email = "team@aiohttp.org"}]
```
Comment (webknjaz) [inline pyproject.toml]: Could you move this into a separate `[[project.maintainers]]` section? Example: https://github.com/cherrypy/cheroot/blob/587d1fa/pyproject.toml#L57-L59.

File: pyproject.toml [hunk 14]
```diff
@@ -1,10 +1,66 @@
...
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
+  "Topic :: Internet :: WWW/HTTP",
+]
+requires-python = ">= 3.9"
+dynamic = ["version", "dependencies", "optional-dependencies"]
```
Comment (webknjaz) [inline pyproject.toml]: Let's have this multi-lined and ordered:
```suggestion
dynamic = [
  "dependencies",
  "optional-dependencies",
  "version",
]
```

File: pyproject.toml (hunk 3 above)
Comment (webknjaz) [reply pyproject.toml]: Is the wildcard needed?
```suggestion
include = [
  "aiohttp",
]
```
Or do we need something like this?
```suggestion
include = [
  "aiohttp",
  "aiohttp.*",
]
```

Personally, though, I'd prefer exclusions over inclusions here.

File: setup.cfg (hunk 4 above)
Comment (webknjaz) [reply setup.cfg:50]: Ack. Marking the thread as resolved.

File: setup.cfg (hunk 1 above)
Comment (webknjaz) [reply setup.cfg:2]: Since we agreed on a follow-up, I'm marking this as resolved for now.

**Comment by webknjaz** [comment]
@cdce8p short of a few cosmetic requests, this seems almost ready.

File: MANIFEST.in
Comment (webknjaz) [reply MANIFEST.in:1]: Marking this thread as resolved.

File: pyproject.toml (hunk 9 above)
Comment (cdce8p) [reply pyproject.toml:15]: > Could you add a code comment with the update guidelines, then?

👍🏻

> And maybe multi-line this by separating it into `[project.license]`: I like it when it's sparse and everything is on dedicated lines as it helps with nicer diffs over time.

I'd prefer to leave it as an inline dict as this simplifies the diff later.
```diff
-license     = {text = "Apache-2.0 AND MIT"}
+license     = "Apache-2.0 AND MIT"
```

File: pyproject.toml (hunk 3 above)
Comment (cdce8p) [reply pyproject.toml]: > Is the wildcard needed?

Technically no since all subpackages would be included as package data anyway. However, I usually still recommend it just in case the package data include / exclude changes.

> Or do we need something like this? [`"aiohttp"`, `"aiohttp.*"`]

Not sure there is an advantage using it over just `aiohttp*`.

> Personally, though, I'd prefer exclusions over inclusions here.

Exclusion is quite easy to get accidentally wrong. Just `exclude = ["examples"]` as it was before wouldn't be enough. Any folder included in the sdist which contains at least one `.py` file would be included / have to be manually excluded (i.e. `docs`, `requirements`, `tests`, `tools`, `vendor`).

--
Overall I'd recommend to stick with `include = ["aiohttp*"]` in this case.

File: setup.cfg (hunk 8 above)
Comment (cdce8p) [reply setup.cfg:19]: Just double checked the metadata and included files with the latest changes. It looks good to me.
```diff
 ...
-Maintainer: aiohttp team <team@aiohttp.org>
-Maintainer-email: team@aiohttp.org
+Maintainer-email: aiohttp team <team@aiohttp.org>
 License: Apache-2.0 AND MIT
-Home-page: https://github.com/aio-libs/aiohttp
+Project-URL: Homepage, https://github.com/aio-libs/aiohttp
 ...

**Comment by cdce8p** [comment]
@webknjaz Pushed another update to address the style changes and also responded to your questions. Please let me know if I missed anything.

File: pyproject.toml (hunk 9 above)
Comment (webknjaz) [reply pyproject.toml:15]: Okay. Marking this chain as resolved.

File: pyproject.toml (hunk 3 above)
Comment (webknjaz) [reply pyproject.toml]: Go with the second code suggestion, then. I don't like that `aiohttp*` would match something that may get in accidentally, like `aiohttp1`.

File: setup.cfg (hunk 8 above)
Comment (webknjaz) [reply setup.cfg:19]: Ack. Marking as resolved.

File: pyproject.toml (hunk 3 above)
Comment (cdce8p) [reply pyproject.toml]: Done. Added your code suggestion. Also double checked it again locally. The `RECORD` file still looks good.

**Comment by cdce8p** [comment]
@webknjaz I just resolved the conflicts. Would love get get this over the finish line, especially as there isn't much left to do here.

**Review by webknjaz — APPROVED** [review]
Thanks!

**Comment by cdce8p** [comment]
Thanks @webknjaz! I'll work on the backports and open a separate PR to move the dependencies as we talked about.

**Comment by webknjaz** [comment]
Thank you!
